### DISCUSSION TITLE
installing fipy and pysparse with python3

### DISCUSSION THREAD

**Discussion by klkuhlm** [usnistgov/fipy#909] [General]
I had to update a system, and I was trying to re-install fipy (ubuntu 20.04) using conda. It seems pysparse is python 2.7 only (has this always been the case?) I am not able to install using the suggested command (which I thought had worked in the not-too-distant past).

`$ conda create --name fipy --channel conda-forge python=3 numpy scipy matplotlib pysparse gmsh
`
It tries many different things, but fails in the end. After some attempts, it seems it is just pysparse that is causing the issues.

searching for pysparse available on conda-forge, I get the following

```
$ conda search pysparse --channel conda-forge
Loading channels: done
# Name                       Version           Build  Channel             
pysparse                     1.3.dev  py27h0c33204_4  conda-forge         
pysparse                     1.3.dev py27h3010b51_1001  conda-forge         
pysparse                     1.3.dev py27h3010b51_1002  conda-forge         
pysparse                     1.3.dev py27h3010b51_1003  conda-forge         
pysparse                     1.3.dev  py27h3010b51_5  conda-forge         
pysparse                     1.3.dev  py27h7eb728f_0  conda-forge         
pysparse                     1.3.dev  py27h7eb728f_1  conda-forge         
pysparse                     1.3.dev  py27h7eb728f_2  conda-forge         
pysparse                     1.3.dev  py27h8db774e_7  conda-forge         
pysparse                     1.3.dev  py27h970eebf_6  conda-forge       
```

Has something changed? Is there a different way to install fipy for python3 (leaving out pysparse?)  

Sorry if I am missing something obvious.

Kris

**Comment by guyer** [comment]
Pysparse has only ever been available for Python 2.7. 

On Ubuntu, you should be able to do
```shell
conda create --name <MYFIPYENV> --channel conda-forge python=3 fipy gmsh
```

If you want to install a develop version of FiPy (GitHub checkout for instance), then follow our instructions for setting up a [development environment](https://www.ctcms.nist.gov/fipy/INSTALLATION.html#development-environment).

**Comment by klkuhlm** [comment]
thank you. this works!  thanks for the link to the instructions for the development version, too.
